### ISSUE TITLE
Custom Error Message for required field with ajv-errors

### ISSUE THREAD

**Issue by THirrrsch** [jarvelov/vue-form-json-schema#67]
### Problem

Hi!

In order to display custom error messages, I'm using `ajv-errors` as you suggest in [your documentation](https://jarvelov.gitbook.io/vue-form-json-schema/api/vue-form-json-schema/options/ajv#extend-ajv-instance). Everything works fine so far with one exception: required fields don't get their custom error message displayed. 

Let me use an example to show the behavior: 
```javascript
{
    schema: 'http://json-schema.org/draft-07/schema#',
    type: 'object',
    required: ['name'],
    errorMessage: {
        required: 'Custom required Error',
    },
    properties: {
        name: {
            type: 'string',
        },
    },
}
```
Using `ajv.validate()` directly, this would work. But since `getVfjsValidationSchema()` validates every sub-schema on its own, the `errorMessage` part is not not passed to `ajv` when validating the `name: { type: 'string' }` schema.
From the example above, the function `getVfjsValidationSchema` creates the following schema for validation: 

```javascript
{
    type: 'object',
    properties: {
        name: {
            type: 'string',
        },
    },
}
```

---

### Proposal

Inside the `getVfjsValidationSchema` function I suggest the following change: 
```javascript
if (isRequired) {
  set(schema, required, [path]);

  // IF   ajv-errors plugin is present (errorMessage keyword is added)
  // AND  fieldSchema has a custom errorMessage definition:
  // THEN add that errorMessage object to the root level of this schema
  //      so that ajv can handle this properly with the ajv-errors plugin
  if (this.ajv.RULES.keywords.errorMessage && fieldSchema.errorMessage) {
    set(schema, "errorMessage", fieldSchema.errorMessage);
  }
}
```

This change makes the following schema work as expected:
```javascript
{
    schema: 'http://json-schema.org/draft-07/schema#',
    type: 'object',
    required: ['name'],
    properties: {
        name: {
            type: 'string',
            errorMessage: {
                required: 'Custom required Error',
            },
        },
    },
}
```
Side note: `ajv-errors` usually expects the `errorMessage` object to be on the schema's root level. However, to make it work with `vue-form-json-schema` the `errorMessage` object needs to be defined inside each of the properties. Thus, with this change the custom error message for `required` can be defined the same way like the other validation keywords, see example below.

```javascript
{
    schema: 'http://json-schema.org/draft-07/schema#',
    type: 'object',
    required: ['name'],
    properties: {
        name: {
            type: 'string',
            minLength: 4,
            errorMessage: {
                required: 'Custom required Error',
                minLength: 'Custom minLength Error',
            },
        },
        age: {
            type: 'number',
            minimum: 43,
            errorMessage: {
                minimum: 'Custom minimum Error',
            },
        },
    },
}
```

**Comment by jarvelov** [comment]
Hi!

Thank you for bringing this to my attention. This is clearly a bug and something which should be fixed. I'm on vacation right now and won't be able to look into it until next week. At first glance your suggested fix looks good! I would (most probably) accept it in a PR if you would like to submit it, giving credit is important and I'll nonetheless mention you in the release message.

**Comment by THirrrsch** [comment]
Hi!

I kept working on that together with @Fuasmattn and I think we finalized our solution. There's a small PR for that now:
#68 

We also found out that your implementation of `getVfjsPropertiesRequired` doesn't work 100% correctly with deeply nested fields. In such cases, `error.dataPath` could be something like `/complexObj/sub1/sub2/sub3`, which would make `propertyPath` become `complexObj/sub1.sub2`. So we replace all occurences of slash with a dot inside the construction of `parent`.

**Comment by jarvelov** [comment]
Awesome work! I have not encountered the slash issue previously, thanks for patching that too! I'm still on vacation but I will test out the pull requests this the first thing I do next week.

**Comment by THirrrsch** [comment]
Hi @jarvelov !

Thank you for considering and reviewing our changes. We're pretty sure that they fix some issues, but obviously you are way more into the details. Are there any findings on your side already, or is there something we can do further?

**Comment by jarvelov** [comment]
Hi @THirrrsch 

I have finished reviewing PR #62 and will proceed with your PR, #68, tonight. Unfortunately work has been pretty demanding these last few days and I only got some free time yesterday to look over the PRs. I do not see any problems with your PR so I hope to have both it and #62 merged and released this weekend in `v2.8.0`.

**Comment by jarvelov** [comment]
PR #68 has now been released in `v2.8.1`!
